cpp/libcudf_kafka/include/cudf_kafka/kafka_consumer.hpp

```diff
@@ -146,6 +146,13 @@ class kafka_consumer : public cudf::io::datasource {
    */
   int64_t get_committed_offset(std::string const &topic, int partition);
 
+  /**
+   * @brief Query the Kafka broker for the assigned topic Metadata
+   *
+   * @return Map of Kafka topics with their corresponding topic partition values.
+   */
+  const std::map<std::string, std::vector<int32_t>> list_topics();
```

Comment:
Higher level question: Is this something that we should add APIs for in libcudf / cudf Python or we should expect people to use `librdkafka` / `confluence-kafka-python` for?

Reply:
@kkraus14 agreed. See https://github.com/rapidsai/cudf/pull/6220#discussion_r488084824